pulsar-client-cpp/lib/MessageIdImpl.h

```diff
@@ -22,15 +22,21 @@
 #include <cstdint>
 
 namespace pulsar {
+const static std::string emptyString = "";
 
 class MessageIdImpl {
    public:
-    MessageIdImpl() : ledgerId_(-1), entryId_(-1), partition_(-1), batchIndex_(-1) {}
+    MessageIdImpl() : ledgerId_(-1), entryId_(-1), partition_(-1), batchIndex_(-1), topicName_(emptyString) {}
```

Comment:
no need to use `emptyString`. initializing with `topicName_()` will have same effect